update-completed-sprint-on-issue-closed/workflow.yml

```diff
@@ -0,0 +1,90 @@
+name: Update CompletedSprint on Issue Closed
+
+on:
+  issues:
+    types: [closed]
+  pull_request:
+    types: [closed]
+
+
+jobs:
+  update_completed_sprint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get latest completed sprint
+        id: get_latest_completed_sprint
+        uses: actions/github-script@v5
+        with:
+          script: |
+            const eventName = context.eventName;
+            const url = eventName === 'issues' || eventName === 'issue_comment' ?
+              payload.issue.html_url :
+              payload.pull_request.html_url;
+            const { data } = await github.graphql(`
+              query {
+                resource(url: "${url}") {
+                  ... on ${eventName.startsWith('issue') ? 'Issue' : 'PullRequest'} {
+                    id,
+                    projectV2(number: 37) {
```

Comment:
💡 You could make the `37` an input parameter. Then the calling repository workflows can provide the value.

You do that by adding the following to the top of this file:

```yaml
  workflow_call:
    inputs:
      project_id:
        description: 'Project ID of the Project the Issue is in'
        type: 'string'
        required: true
```

In the github-script you can reference the input with:

```
${{ inputs.project_id }}
```

In the workflow that calls this workflow, you can pass the value with:

```yaml
jobs:
  update-completed-sprint-on-issue-closed:
    uses: stellar/actions/.github/workflows/update-completed-sprint-on-issue-closed.yml@main
    with:
      project_id: 37
```

Reply:
💡 Alternatively, you could query the GitHub API and get the list of projects the issue is associated with:
https://docs.github.com/en/graphql/reference/objects#issue

Reply:
Yeah, personal oauth keys aren't allowed to list projects (owned by the org), so testing any more advanced graphql queries is quite difficult. Trying to figure out a good way to test this, tbh.

Reply:
Opted to query based on the project name, which seems more intuitive to use.